app/src/controllers/bucketPermission.js

```diff
@@ -0,0 +1,114 @@
+const Problem = require('api-problem');
+
+const errorToProblem = require('../components/errorToProblem');
+const {
+  addDashesToUuid,
+  mixedQueryToArray,
+  getCurrentIdentity
+} = require('../components/utils');
+const { NIL: SYSTEM_USER } = require('uuid');
+const { bucketPermissionService, userService } = require('../services');
+
+const SERVICE = 'BucketPermissionService';
+
+/**
+ * The Permission Controller
+ */
+const controller = {
+  /**
+   * @function searchPermissions
+   * Searches for bucket permissions
+   * @param {object} req Express request object
+   * @param {object} res Express response object
+   * @param {function} next The next callback function
+   * @returns {function} Express middleware function
+   */
+  async searchPermissions(req, res, next) {
+    try {
+      const bucketIds = mixedQueryToArray(req.query.bucketId);
+      const userIds = mixedQueryToArray(req.query.userId);
+      const response = await bucketPermissionService.searchPermissions({
+        bucketId: bucketIds ? bucketIds.map(id => addDashesToUuid(id)) : bucketIds,
+        userId: userIds ? userIds.map(id => addDashesToUuid(id)) : userIds,
+        permCode: mixedQueryToArray(req.query.permCode)
+      });
+      res.status(200).json(response);
+    } catch (e) {
+      next(errorToProblem(SERVICE, e));
+    }
+  },
+
+  /**
+   * @function listPermissions
```

Comment:
an idea: we could have just the `searchPermissions()` method above. same with objectPermission listPermissions() method . feels a bit like duplication.